src/rdo.rs

```diff
@@ -859,14 +897,16 @@ pub fn rdo_partition_decision(
         //pmv = best_pred_modes[0].mvs[0];
 
         assert!(best_pred_modes.len() <= 4);
-        let bs = bsize.width_mi();
-        let hbs = bs >> 1; // Half the block size in blocks
-        let partitions = [
+
+        let hbsw = subsize.width_mi(); // Half the block size width in blocks
+        let hbsh = subsize.height_mi(); // Half the block size height in blocks
+        let four_partitions = [
           bo,
-          &BlockOffset{ x: bo.x + hbs as usize, y: bo.y },
-          &BlockOffset{ x: bo.x, y: bo.y + hbs as usize },
-          &BlockOffset{ x: bo.x + hbs as usize, y: bo.y + hbs as usize }
+          &BlockOffset{ x: bo.x + hbsw as usize, y: bo.y },
+          &BlockOffset{ x: bo.x, y: bo.y + hbsh as usize },
+          &BlockOffset{ x: bo.x + hbsw as usize, y: bo.y + hbsh as usize }
         ];
+        let partitions = get_sub_partitions(&four_partitions, partition);
 
         let pmv_idxs = partitions.iter().map(|&offset| {
           if subsize > BlockSize::BLOCK_32X32 {
```

Comment:
For splitting into 64x32 or 32x64, I think this should be fixed as : subsize.larger_than(BlockSize::BLOCK_32X32).

Reply:
We can test which way is better.